src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/databaseFeedManager.hpp

```diff
@@ -275,6 +281,16 @@ class TDatabaseFeedManager final : public Observer<nlohmann::json&>
         m_translationsCache =
             std::make_unique<LRUCache<std::string, std::string>>(TPolicyManager::instance().getTranslationLRUSize());
 
+        // This initializes the vendor maps and should be called before any scan or message processing.
+        try
+        {
+            reloadVendorMaps();
+        }
+        catch (const std::exception& e)
+        {
+            logWarn(WM_VULNSCAN_LOGTAG, "Error initializing vendor maps: %s", e.what());
+        }
+
```

Comment:
This does not make sense... We are capturing an exception here, so after executing the constructor in vulnerabilityScannerFacade we ask if the map could be loaded and throw an exception, I prefer that the exception be propagated.

It is important to handle the case of when the content is generated with the testtool, so that this verification is not done.

```suggestion
            reloadVendorMaps();
```

Reply:
Hi @Dwordcito !

I considered the case when the content is being generated, that is why I took this approach.
My intention was that the `databaseFeedManager` doesn't fail at the start if the vendor map was invalid, only after new content is received.
And like vulnerability detector needs the map, I created a method to verify the state of map after the start of `databaseFeedManager`.

>  We are capturing an exception here, so after executing the constructor in vulnerabilityScannerFacade we ask if the map could be loaded and throw an exception

The exception isn't captured in `processMessage()`.

> It is important to handle the case of when the content is generated with the testtool, so that this verification is not done.

This approach requires to add another flag in the vulnerability detector `start()` method because the test tool initiates the module in the same way **modulesd** does.